.github/workflows/stm32build.yml

```diff
@@ -78,47 +78,90 @@ jobs:
       pull-requests: read
       contents: write
     # If no changes has been made to projects above do not build
-    if: ${{ needs.changes.outputs.packages != '[]' && needs.changes.outputs.packages != '' }} 
+    if: ${{ needs.changes.outputs.packages != '[]' && needs.changes.outputs.packages != '' }}
     strategy:
       matrix:
         # Parse JSON array containing names of all filters matching any of changed files
         package: ${{ fromJSON(needs.changes.outputs.packages) }}
 
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3                         # gets file environment with submodule included for libraries
-      with:
-        submodules: 'true'
-    - uses: carlosperate/arm-none-eabi-gcc-action@v1    # Setup the compiler and compile the program
-    - run: make
-      working-directory: 'STM32/${{ matrix.package }}/'
-    - uses: actions/upload-artifact@v3                  # Upload the .elf and .bin file under the summary tab
-      with: 
-        name: ${{ matrix.package }}
-        path: | 
-          STM32/${{ matrix.package }}/build/${{ matrix.package }}.elf
-          STM32/${{ matrix.package }}/build/${{ matrix.package }}.bin
-    - uses: copenhagenatomics/mono_release_on_push_action@master
-      id: release-on-push
-      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # tag only pushes to the main branch
-      with:
-        bump_version_scheme: minor
-        use_github_release_notes: true
-        tag_prefix: "${{ matrix.package }}-"
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-    - name: Read PCB version
-      id: pcbversion
-      run: echo "version=$(head -1 STM32/${{ matrix.package }}/pcbversion)" >> $GITHUB_OUTPUT
-    - name: Release
-      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # upload only pushes to the main branch
-      uses: softprops/action-gh-release@v0.1.15
-      with:
-        files: |
+      # gets file environment with submodule included for libraries
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+      # Setup the compiler and compile the program
+      - uses: carlosperate/arm-none-eabi-gcc-action@v1
+      # Generate the next release tag for including in the build. Note: dry_run. It doesn't actually
+      # Release in case something goes wrong in the build
+      - uses: copenhagenatomics/mono_release_on_push_action@master
+        id: release-on-push-1
+        with:
+          bump_version_scheme: minor
+          dry_run: "true"
```

Comment:
This part is "dry_run" because:
1. We need a version number to build FW with version number built in
2. if something goes wrong with the build, we don't want to generate a tag&release (and mono-release-on-push has a side effect of generating a release)

It does say this in the comment above this step, but perhaps it needs to be spelt out more